### PR TITLE
Multiple municipality filter, implemented on agenda-items & sessions.index

### PR DESCRIPTION
**Pull request by Denperidge**
Note: this PR relies on #22 , but has been separated because it influences how the queries in the routes/controllers are made

But yes! It works! Kind of? It should be noted that - either through the BNB caching or ember-data caching - the first selected municipality and the no-municipality ones are shown first, so you've got to scroll a little bit. But this should hopefully fix itself down the road with the chronological sort

### CODE REVIEW

**Comment by aliokan** [comment]
@Denperidge Can you handle the branch conflict before I start review? :pray:

File: app/routes/agenda-items.ts [hunk 1]
```diff
@@ -28,7 +30,7 @@ const getQuery = ({
     "session.governing-body.administrative-unit",
     "session.governing-body.administrative-unit.location",
   ].join(","),
-  municipality: municipality ? municipality : undefined,
+  //municipality: locationIds ? locationIds : undefined,
```
Comment (aliokan) [inline app/routes/agenda-items.ts:33]: **nitpick:** Can you remove commented code in order to keep code clean :pray:

File: app/routes/agenda-items.ts [hunk 2]
```diff
@@ -135,35 +140,45 @@ export default class AgendaItemsRoute extends Route {
...
       return model;
     }
     this.keywordStore.keyword = params.keyword || "";
-    this.municipality = params.municipality || "";
+    this.municipalityLabels = params.municipalityLabels || "";
     this.sort = params.sort || "";
     this.plannedStartMin = params.plannedStartMin || "";
     this.plannedStartMax = params.plannedStartMax || "";
 
+    /*
```
Comment (aliokan) [inline app/routes/agenda-items.ts:156]: **nitpick:** Can you remove commented code in order to keep code clean 🙏

File: app/routes/sessions/index.ts [hunk 3]
```diff
@@ -2,17 +2,21 @@ import Store from "@ember-data/store";
 import Route from "@ember/routing/route";
 import { service } from "@ember/service";
 import { tracked } from "@glimmer/tracking";
+import MunicipalityListService from "frontend-burgernabije-besluitendatabank/services/municipality-list";
+import { seperator } from "frontend-burgernabije-besluitendatabank/components/filter-sidebar/select-multiple-filter";
```
Comment (aliokan) [inline app/routes/sessions/index.ts:6]: **Issue:** A route shouldn't be dependents of a component. What about moving `seperator` in a constants file and use it in both `select-multiple-filter` component and `session.index` route?

File: app/services/municipality-list.ts [hunk 4]
```diff
@@ -28,4 +28,20 @@ export default class MunicipalityListService extends Service {
...
     }
+
+
+    async getLocationIdsFromLabels(labels: Array<string>): Promise<Array<string>> {
+        const locationIds : Array<string> = [];
+        const municipalities = await this.municipalities();
+        
+        for (let i = 0; i < labels.length; i++) {
+          let label = labels[i];
+          let municipality = municipalities.find((municipality) => municipality.label == label)
```
Comment (aliokan) [inline app/services/municipality-list.ts:39]: **nitpick:** strict equality (`===`) is safer in JS world.

File: app/components/filter-sidebar/select-multiple-filter.ts [hunk 5]
```diff
@@ -0,0 +1,59 @@
...
+
+            this.args.options.then((haystack: [{[key:string]: any}]) => {
+                if (!queryParam) return;  // TypeScript was being wacky
+
+                let results: Array<{[key:string]: any}> = [];
+
+                for (let i = 0; i < needles.length; i++) {
+                    let needle = needles[i];
+                    console.log("Search for: " + needle)
+                    let found = haystack.find((value) => get(value, searchField) == needle);
```
Comment (aliokan) [inline app/components/filter-sidebar/select-multiple-filter.ts:32]: **nitpick:** strict equality (`===`) is safer in JS world.

**Review by aliokan — CHANGES_REQUESTED** [review]
Some comments added

File: app/components/filter-sidebar/filter.ts [hunk 6]
```diff
@@ -7,12 +7,18 @@ interface ArgsInterface {
...
+
     searchField: string;
+    options: any;
 }
 
 export default class FilterComponent extends Component<ArgsInterface> {
     @service declare router: RouterService;
 
+    getQueryParam(param: string) {
+        return get(this.router.currentRoute.queryParams, param);
```
[the remark this answers is not in the document]
Comment (Denperidge) [reply app/components/filter-sidebar/filter.ts:19]: Fix in next commit!

File: app/components/filter-sidebar/select-multiple-filter.hbs [hunk 7]
```diff
@@ -0,0 +1,29 @@
+<FilterSidebar::Filter @id={{@id}} @label={{@label}}>
+    <PowerSelectMultiple
+      style="width: 100%;"
```
[the remark this answers is not in the document]
Comment (Denperidge) [reply app/components/filter-sidebar/select-multiple-filter.hbs:3]: Changed it to an appuniverse class!

[the remark this answers is not in the document]
Comment (Denperidge) [reply app/components/filter-sidebar/select-multiple-filter.hbs:3]: Update: I realised you might've meant that the full width specification may not be needed, which is the case! Removed style & class in next commit

File: app/components/filter-sidebar/select-multiple-filter.ts [hunk 8]
```diff
@@ -0,0 +1,59 @@
...
+    @action
+    async inserted() {
+        // The queryParam is an array of searchField values, joined by seperator
+        // On the page load, we can use this to load in values
+        let queryParam = this.getQueryParam(this.args.queryParam);
+        if (queryParam) {
+            const needles = queryParam.split(seperator);
+            const searchField = this.args.searchField;
+
+            this.args.options.then((haystack: [{[key:string]: any}]) => {
```
[the remark this answers is not in the document]
Comment (Denperidge) [reply app/components/filter-sidebar/select-multiple-filter.ts:24]: Fixed!

File: app/components/filter-sidebar/select-multiple-filter.ts [hunk 9]
```diff
@@ -0,0 +1,59 @@
...
+                if (!queryParam) return;  // TypeScript was being wacky
```
[the remark this answers is not in the document]
Comment (Denperidge) [reply app/components/filter-sidebar/select-multiple-filter.ts:25]: Implicitly solved with the await change!

File: app/controllers/agenda-items.ts [hunk 10]
```diff
@@ -29,15 +30,19 @@ export default class AgendaItemsController extends Controller {
   async loadMore() {
     if (this.model && !this.isLoadingMore) {
       this.isLoadingMore = true;
+
+      let locationIds = await this.municipalityList.getLocationIdsFromLabels(this.municipalityLabels.split(seperator));
+      console.log(locationIds)
```
[the remark this answers is not in the document]
Comment (Denperidge) [reply app/controllers/agenda-items.ts:35]: Fixed!

File: app/routes/agenda-items.ts (hunk 2 above)
Comment (Denperidge) [reply app/routes/agenda-items.ts:156]: Done!

File: app/routes/agenda-items.ts (hunk 1 above)
Comment (Denperidge) [reply app/routes/agenda-items.ts:33]: Done!

File: app/routes/sessions/index.ts (hunk 3 above)
Comment (Denperidge) [reply app/routes/sessions/index.ts:6]: Good idea! Created `helpers/constants.ts`. Don't know if that's the optimal location though, so feel free to move it if desired

File: app/services/municipality-list.ts (hunk 4 above)
Comment (Denperidge) [reply app/services/municipality-list.ts:39]: Done!

File: app/components/filter-sidebar/select-multiple-filter.ts (hunk 5 above)
Comment (Denperidge) [reply app/components/filter-sidebar/select-multiple-filter.ts:32]: Done!

**Review by aliokan — APPROVED** [review]
Looks good, LGTM! :rocket:

**Pull request merged**